cobc/flag.def

```diff
@@ -104,16 +104,32 @@ CB_FLAG_NQ (1, "default-file-colseq", CB_FLAG_GETOPT_DEFAULT_FILE_COLSEQ,
 	_("  -fdefault-file-colseq=[ASCII|EBCDIC|NATIVE]\tdefine default file collating sequence\n"
 	  "                        * default: NATIVE"))
 
-/* Binary flags */
+/* Binary flags, partially with suppressed help */
 
 CB_FLAG (cb_flag_sql_xfd, 1, "sql",
 	_("  -fsql                 process the XFD directives"))
 
-/* Flags with suppressed help */
-
-CB_FLAG (cb_flag_winmain, 0, "winmain",
+#ifdef _WIN32
+#define COB_SHOW 1
+#else
+#define COB_SHOW 0
+#endif
+CB_FLAG (cb_flag_winmain, COB_SHOW, "winmain",
 	_("  -fwinmain             generate WinMain instead of main when compiling\n"
 	  "                        as executable"))
+#undef COB_SHOW
+
+#ifdef HAVE_ATTRIBUTE_CONSTRUCTOR
+#define CB_FLAG_M CB_FLAG_ON
+#define COB_SHOW 1
+#else
+#define CB_FLAG_M CB_FLAG
+#define COB_SHOW 0
+#endif
+CB_FLAG_M (cb_flag_use_constructor, COB_SHOW, "use-constructor",
+	_("  -fuse-constructor     generate internal one-time code via constructor"))
+#undef CB_FLAG_M
+#undef COB_SHOW
```

Comment:
Hm, I don't remember that commit... and _WIN32 is not only define for _MSC_VER but on all Win32 but Cygwin...

Reply:
That's from SVN commit 5245. I could change `_WIN32` to `_MSC_VER` if that's appropriate (I'll check).

Reply:
Well, I think `_WIN32` is right in this context. If we're on Windows, no matter if it is MSVC or not, we want to be able to use WinMain (to make a Windows GUI application). I checked and WinMain works both on Cygwin and MSYS2 (all environments) (additionally, it requires the `-mwindows` gcc/clang flag if we don't want to display any console).